test/e2e/kubectl/kubectl.go

```diff
@@ -1481,6 +1481,7 @@ metadata:
 			time.Sleep(2500 * time.Millisecond) // ensure that startup logs on the node are seen as older than 1s
 			recentOut := framework.RunKubectlOrDie(ns, "logs", podName, containerName, nsFlag, "--since=1s")
 			recent := len(strings.Split(recentOut, "\n"))
+			time.Sleep(2500 * time.Millisecond) // ensure that enough logs on the node are created past previous call
```

Comment:
@eddiezane I don't think this is actually addressing the issue because it appears this is trying to test that there were more logs (i.e. the startup logs) written before the `recent`, so waiting longer after reading the `recent` logs and then checking the `older` is not testing that the startup logs are included in `older` but not `recent`. See above:
```
// ensure that startup logs on the node are seen as older than 1s
```

Do you read differently?

Reply:
@hasheddan this whole test is confusing.

```
gomega.Expect(recent).To(gomega.BeNumerically("<", older), "expected recent(%v) to be less than older(%v)\nrecent lines:\n%v\nolder lines:\n%v\n", recent, older, recentOut, olderOut)
```

I believe it's asserting that there are less `recent` (logs created 1 second ago `--since=1s`) than there are `older` (logs created 24 hours ago `--since=24h`). Which should be the case. The flake was occurring when there isn't enough time between the `recent` and `older` (because they are both basically starting from 0 logs).

Thoughts?

Reply:
Hmm shouldn't the first wait (https://github.com/kubernetes/kubernetes/pull/93763/files#diff-fde2ce0a73639d19347a5ea6fe001ba6R1481) take care of making it true that there are more logs in the last 24h than in the last 1s? This is why this flake is concerning to me because it _seems_ like the test logic is sound. If anything, it looks like we should bump the first wait to a longer period.

Reply:
@hasheddan I _think_ that the test starts with 0 logs (the pod is spun up at the very start of this test) so that initial wait doesn't create more `older` logs. The test failures looked like they were because the two vars were equal at a very low number (like 5).

Reply:
@eddiezane I believe the purpose of the first wait is to make there be older logs than 1s. So if the pod starts up and starts producing logs, then we wait 2.5 seconds, if any logs are produced in the first 1.5 seconds they should be in `older` but not in `recent`. Therefore, extending the first wait would increase the likelihood that logs are produced before the last `1s` that `recent` looks at.